### PR TITLE
Energy gun overlay fixes and icon reversions

### LINKED ISSUE #10730: New gun overlays broke at least one gun
http://i.imgur.com/godG7pt.png

Turned the pulse pistol invisible.

### LINKED ISSUE #10731: New gun overlays broke UI button for gunlights
uiicon

### PR DESCRIPTION
**Pull request by Jordie0608**
Fixes #10730
Fixes #10731

Reverts the icons of the following guns and adds support for non-tiling overlay charge bars
HoS's Laser
Laser
Captain's Laser
Retro Laser
Lasertag

### CODE REVIEW

File: code/_onclick/hud/action.dm [hunk 1]
```diff
@@ -135,12 +135,12 @@
 	var/image/img
 	if(owner.action_type == AB_ITEM && owner.target)
 		var/obj/item/I = owner.target
-		img = image(I.icon, src , I.icon_state)
+		overlays += I.appearance
```
Comment (phil235) [inline code/_onclick/hud/action.dm]: don't you mean += img ?

Comment (Jordie0608) [inline code/_onclick/hud/action.dm]: No; appearance grabs a copy of an atom's entire icon, overlays and underlays included, whereas an image object is limited to just an icon_state

Comment (RemieRichards) [inline code/_onclick/hud/action.dm]: @phil235 , @Jordie0608 is using a 508 feature, the ability to grab internal appearances (cached collections of underlays, overlays, icon, icon_state, and other visual data) in one variable, and swap them/add them to other objects.

**Comment by Firecage** [comment]
Jordie. You have made me love you once again. Thankyou.

**Comment by Steelpoint** [comment]
I fully support this!

**Comment by freerealestate** [comment]
>old
>

>new
>


It's subjective, but I actually preferred some of the new sprites you made. Captain's laser, regular laser, laser cannon and retro laser all look better imo. It's a shame you're reverting them.

**Comment by Anonus** [comment]
This is, indeed, debatable. I'm on the other side, old sprites better new sprites bad.

**Comment by SconesC** [comment]
Restore shaded bar Captain's Laser, restore small tick'd bar HoS Gun

:+1: 

Both are kind of hideous with the current sprites

**Comment by Lumipharon** [comment]
Honestly pips look way shittier then having the bar - on every single gun.

**Comment by Steelpoint** [comment]
I'll do a look through of the guns tomorrow and see if they can be improved with a similar system to the HoS gun.

**Pull request merged**